libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -809,20 +807,35 @@ std::string YulUtilFunctions::resizeDynamicArrayFunction(ArrayType const& _type)
 
 				let oldLen := <fetchLength>(array)
 
-				// Store new length
-				sstore(array, newLen)
+				<?isBytes>
+					switch lt(newLen, 32)
+					case 0 {
+						sstore(array, add(mul(newLen, 2), 1))
+					}
+					default {
+						sstore(array, mul(newLen, 2))
+					}
```

Comment:
```suggestion
					sstore(array, or(shl(newLen, 1), iszero(lt(newLen, 32))))
```
or is that too cryptic :-)?

Reply:
I decided to move resizing bytes array to its own PR